rollup/rollup_sync_service/rollup_sync_service.go

```diff
@@ -422,41 +422,57 @@ func (s *RollupSyncService) decodeChunkBlockRanges(txData []byte) ([]*rawdb.Chun
 	return nil, fmt.Errorf("unexpected method name: %v", method.Name)
 }
 
-// validateBatch verifies the consistency between the L1 contract and L2 node data.
+// validateBatch verifies the consistency between L1 contract and L2 node data.
+// It performs the following checks:
+// 1. Recalculates the batch hash locally
+// 2. Compares local state root, withdraw root, and batch hash with L1 data (for the last batch only when "finalize by bundle")
+//
 // The function will terminate the node and exit if any consistency check fails.
-// It returns the number of the end block, a finalized batch meta data, and an error if any.
-func validateBatch(event *L1FinalizeBatchEvent, parentBatchMeta *rawdb.FinalizedBatchMeta, chunks []*encoding.Chunk, chainCfg *params.ChainConfig, stack *node.Node) (uint64, *rawdb.FinalizedBatchMeta, error) {
+//
+// Parameters:
+// - event: L1 finalize batch event data
+// - parentBatchMeta: metadata of the parent batch
+// - chunks: slice of chunk data for the current batch
+// - chainCfg: chain configuration to identify the codec version
+// - stack: node stack to terminate the node in case of inconsistency
+// - lastBatch: boolean indicating if this is the last batch in a bundle
+//
+// Returns:
+// - uint64: the end block height of the batch
+// - *rawdb.FinalizedBatchMeta: finalized batch metadata
+// - error: any error encountered during validation
+//
+// Note: This function is compatible with both "finalize by batch" and "finalize by bundle" methods.
+// In "finalize by bundle", only the last batch of each bundle is fully verified.
```

Comment:
will be great if we can also add this to the PR description

Reply:
make sense. added.